### PR TITLE
Allow listing of inactive offerings

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR fixes https://github.com/apache/cloudstack/issues/8817 by allowing listing of Inactive offering ids to the user and changing state of service offering & disk offering using the update commands.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan test

File: engine/schema/src/main/resources/META-INF/db/views/cloud.service_offering_view.sql [hunk 1]
```diff
@@ -108,7 +109,5 @@ FROM
         LEFT JOIN
     `cloud`.`service_offering_details` AS `vsphere_storage_policy` ON `vsphere_storage_policy`.`service_offering_id` = `service_offering`.`id`
         AND `vsphere_storage_policy`.`name` = 'storagepolicy'
-WHERE
-        `service_offering`.`state`='Active'
```
Comment (sureshanaparti) [inline engine/schema/src/main/resources/META-INF/db/views/cloud.service_offering_view.sql:112]: add this view changes in upgrade path

**Review by sureshanaparti — CHANGES_REQUESTED** [review]
@vishesh92 Please add the view changes in upgrade path, Rest changes LGTM.

The current listing behavior (with 'Active' state) is maintained in API / UI ?

**Comment by DaanHoogland** [comment]
> @vishesh92 Please add the view changes in upgrade path, Rest changes LGTM.
> 
> The current listing behavior (with 'Active' state) is maintained in API / UI ?

Views are now updated from their individual view sql files @sureshanaparti . These are processed during upgrade.

**Comment by sureshanaparti** [comment]
> > @vishesh92 Please add the view changes in upgrade path, Rest changes LGTM.
> > The current listing behavior (with 'Active' state) is maintained in API / UI ?
> 
> Views are now updated from their individual view sql files @sureshanaparti . These are processed during upgrade.

Thanks for confirming @DaanHoogland

**Review by sureshanaparti — APPROVED** [review]
code LGTM

File: ui/src/config/section/offering.js [hunk 2]
```diff
@@ -89,15 +90,33 @@ export default {
...
+        groupAction: true,
+        popup: true,
+        show: (record) => { return record.state !== 'Active' },
+        groupMap: (selection) => { return selection.map(x => { return { id: x, state: 'Active' } }) }
       }, {
         api: 'deleteServiceOffering',
-        icon: 'delete-outlined',
-        label: 'label.action.delete.service.offering',
-        message: 'message.action.delete.service.offering',
+        icon: 'pause-circle-outlined',
```
Comment (DaanHoogland) [inline ui/src/config/section/offering.js:112]: why would you want to change the icon for delete, @vishesh92 ?

**Review by DaanHoogland — CHANGES_REQUESTED** [review]
I think something went wrong @vishesh92 . the delete button seems gone on the compute-/system-/diskoffering quick menus

File: ui/src/config/section/offering.js (hunk 2 above)
Comment (vishesh92) [reply ui/src/config/section/offering.js:112]: Because it's not actually deleting the service offering. Just changing the state to inactive.

**Review by DaanHoogland — APPROVED** [review]
cltgm and tested, there is some pre-existing issue with error messages when trying to delete a default system service offering.

**Comment by DaanHoogland** [comment]
for the error message issue see #8882

**Pull request merged**